shared/src/ledger/masp.rs

```diff
@@ -200,3 +246,1044 @@ pub fn get_params_dir() -> PathBuf {
         masp_proofs::default_params_folder().unwrap()
     }
 }
+
+#[async_trait]
+pub trait ShieldedUtils : Sized + BorshDeserialize + BorshSerialize + Default + Clone {
+    async fn query_storage_value<T: Send>(
+        &self,
+        key: &storage::Key,
+    ) -> Option<T>
+    where
+        T: BorshDeserialize;
+
+    async fn query_epoch(&self) -> Epoch;
+
+    fn local_tx_prover(&self) -> LocalTxProver;
+
+    fn load(self) -> std::io::Result<ShieldedContext<Self>>;
+
+    fn save(&self, ctx: &ShieldedContext<Self>) -> std::io::Result<()>;
+
+    async fn query_conversion(
+        &self,
+        asset_type: AssetType,
+    ) -> Option<(
+        Address,
+        Epoch,
+        masp_primitives::transaction::components::Amount,
+        MerklePath<Node>,
+    )>;
```

Comment:
Also here, does this mean that we should define this in a web/WASM specific way?

```rust
    async fn query_conversion(
        &self,
        asset_type: AssetType,
    ) -> Option<(
        Address,
        Epoch,
        masp_primitives::transaction::components::Amount,
        MerklePath<Node>,
    )>;
```